signature-aggregator/aggregator/aggregator.go

```diff
@@ -249,7 +250,40 @@ func (s *SignatureAggregator) CreateSignedMessage(
 	if isL1 {
 		log.Debug("Checking L1 validators for zero balance nodes")
 		fetchL1Validators := func(subnetID ids.ID) ([]platformvmapi.APIL1Validator, error) {
-			return s.pChainClient.GetCurrentL1Validators(ctx, subnetID, nil, s.pChainClientOptions...)
+			validators, err := s.pChainClient.GetCurrentValidators(ctx, subnetID, nil, s.pChainClientOptions...)
+			if err != nil {
+				log.Error(
+					"Failed to fetch current L1 validators",
+					zap.String("subnetID", subnetID.String()),
+					zap.Error(err),
+				)
+				return nil, fmt.Errorf("failed to fetch current L1 validators: %w", err)
+			}
+
+			l1Validators := make([]platformvmapi.APIL1Validator, 0, len(validators))
```

Comment:
we don't have tests for this functionality yet, but we should at the very least add unit tests for it. Ideally E2E but creating underfunded nodes in could be tricky with tmpnet.